website/project/model.py

```diff
@@ -1896,6 +1906,10 @@ def root(self):
             return self
 
     @property
+    def registrations(self):
+        return self.node__registrations.find(Q('archiving', 'eq', False))
```

Comment:
I don't really see a performance bottleneck here, but would be interested if you see one.

The ```self.node__registrations``` should be fast, and the ```.find``` is just a filter on that queryset.